x/supplier/keeper/msg_server_stake_supplier.go

```diff
@@ -105,9 +113,10 @@ func (k msgServer) StakeSupplier(ctx context.Context, msg *types.MsgStakeSupplie
 		supplier.UnstakeSessionEndHeight = sharedtypes.SupplierNotUnstaking
 	}
 
-	// MUST ALWAYS stake or upstake (> 0 delta)
-	if coinsToEscrow.IsZero() {
-		err = types.ErrSupplierInvalidStake.Wrapf("Signer %q must escrow more than 0 additional coins", msg.Signer)
+	// MUST ALWAYS stake or upstake (>= 0 delta)
+	// TODO_CONSIDERATION: Should we allow stake decrease down to min stake?
+	if coinsToEscrow.IsNegative() {
+		err = types.ErrSupplierInvalidStake.Wrapf("Signer %q must be higher or equal to the current stake", msg.Signer)
```

Comment:
The error message confused me on my first pass. Wdyt about something like:

```suggestion
		err = types.ErrSupplierInvalidStake.Wrapf("Supplier signer %q stake (%s) must be greater than or equal to the current stake (%s)", msg.Signer, msg.GetStake(), currSupplierStake)
```